app/src/main/kotlin/com/simplemobiletools/clock/fragments/AlarmFragment.kt

```diff
@@ -72,13 +90,14 @@ class AlarmFragment : Fragment(), ToggleAlarmInterface {
     }
 
     private fun setupAlarms() {
-        alarms = context?.dbHelper?.getAlarms() ?: return
-        if (context?.getNextAlarm()?.isEmpty() == true) {
+        val context = context ?: return
+        alarms = context.dbHelper.getAlarms(context.config.alarmSort)
+        if (context.getNextAlarm().isEmpty()) {
             alarms.forEach {
                 if (it.days == TODAY_BIT && it.isEnabled && it.timeInMinutes <= getCurrentDayMinutes()) {
                     it.isEnabled = false
                     ensureBackgroundThread {
-                        context?.dbHelper?.updateAlarmEnabledState(it.id, false)
+                        context.dbHelper.updateAlarmEnabledState(it.id, false)
```

Comment:
this run on a background thread, context could be null by then. Avoid changing unrelated things please, it is done like this for a reason.

Reply:
That is still checked, in line 93 the function returns if `context` is `null` (the same behavior as before). I thought this change would make it a bit easier to read, but if you disagree I can change it back and just add the necessary sorting.

Reply:
you check it above, but like 100 is already on a background thread and context could become null meanwhile

Reply:
You are right, I'm sorry I didn't realize that. I reverted the changes there now and just left the sort parameter itself in, I think that should cover it.